SIP-028: sBTC Signer Criteria (#186)
stacksgov/sips: sips/sip-028/sip-028-sbtc_peg.md

> # SIP-028: sBTC Signer Criteria

Reviewer: ```suggestion
# Preamble
```
SIP should begin with Preamble, which includes Title

Reviewer 2: @aldur @andrerserrano

Reviewer 2: i don't this is resolved, it should be in the format like:
```
# Preamble

SIP Number: 028
...
```

Reviewer 2: i stand corrected, i had an outdated view

**SIP Number:** 028  
**Title:** Signer Criteria for sBTC, A Decentralized and Programmable Asset Backed 1:1 with BTC  

> **Consideration:** Technical, Economics  

Reviewer: ```suggestion
**Consideration:** Technical, Economic  
```
According to the SIP Specification, the Consideration should be 'Economic'. Please correct 'Economics' to 'Economic'.

**Type:** Standard  
**Status:** Draft  
**Authors:**  
- Mårten Blankfors ([marten@trustmachines.co](mailto:marten@trustmachines.co))  
- Daniel Jordon ([daniel@trustmachines.co](mailto:daniel@trustmachines.co))
- Friedger Müffke ([friedger@ryder.id](mailto:friedger@ryder.id))  
- Jesus Najera ([jesus@stratalabs.xyz](mailto:jesus@stratalabs.xyz))  
- Jude Nelson ([jude@stacks.org](mailto:jude@stacks.org))  
- Tycho Onnasch ([tycho@zestprotocol.com](mailto:tycho@zestprotocol.com))  
- Andre Serrano ([andre@bitcoinl2labs.com](mailto:andre@bitcoinl2labs.com))  
- Ashton Stephens ([ashton@trustmachines.co](mailto:ashton@trustmachines.co))  
- Joey Yandle ([joey@trustmachines.co](mailto:joey@trustmachines.co))  

> 

Reviewer: Preamble is missing required fields:

Missing Created field. Please add the ISO 8601 date when this SIP was created.

Missing License field. Please specify the content license for this SIP.

Missing Sign-off field. Please add this field as placeholder, even if it is not yet filled in entirely.

Missing Layer field in Preamble:
Since this is a Standard-type SIP, the Layer field is required. Please specify the logical layer of the Stacks blockchain affected (e.g., 'Applications' or 'Peer Services' since this is about p2p signer network).

Missing Discussions-To field in Preamble:
Please provide a mailing list or discussion forum where ongoing discussion of the SIP takes place (e.g. sBTC Working Group discussions?)

Reviewer 2: @aldur @andrerserrano

Reviewer 2: ```
Missing Discussions-To field in Preamble:
Please provide a mailing list or discussion forum where ongoing discussion of the SIP takes place (e.g. sBTC Working Group discussions?)
```

is this standard? i'm not seeing a precedence for this requirement. all i can find is this, which suggests it may be required *sometimes*: https://github.com/stacksgov/sips/blob/2ffa3ad9de843faf233174388e4a6ab15bc8dcab/sips/sip-000/sip-000-stacks-improvement-proposal-process.md?plain=1#L128

Reviewer: not a must I think, however sip-000 itself is precedence. 
no harm in including relevant discussions which lead to this SIP standard

Reviewer 2: ok thanks - @andrerserrano let's either add that, or explain why it's not needed for this SIP please.

Author: Added a link to this for the sBTC Working Group Discussions

This SIP outlines but does not describe in technical detail the workings of the first sBTC system. A separate SIP will be written to do so if this SIP successfully activates.

## Introduction

### Glossary

| Term                | Definition                                                                                                                                                            |
|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **SIP-10 Token**    | A token on the Stacks blockchain that adheres to the fungible token standards outlined in [SIP-10](https://github.com/stacksgov/sips/blob/main/sips/sip-010/sip-010-fungible-token-standard.md).                                                                       |
| **sBTC**            | A SIP-10 token on the Stacks Blockchain that can be turned back into BTC on the Bitcoin Blockchain. 1 sBTC is equivalent to 1 BTC on the Bitcoin Blockchain.           |

> | **sBTC operation**  | An smart contract function call that initiates some action from the sBTC protocol.                                                                                                        |

Reviewer: ```suggestion
| **sBTC operation**  | A smart contract function call that initiates some action from the sBTC protocol.                                                                                                        |
```
fix grammar

| **.sbtc contract**  | A smart contract (or a collection of contracts) defining the sBTC token and functions related to it.                                                                    |
| **sBTC Peg Wallet** | The single UTXO holding the entire BTC balance that’s pegged into sBTC. This peg wallet is managed and maintained by the sBTC Signers.                                  |
| **Stacks Signer**   | An entity that receives PoX payouts for stacking their STX tokens and actively participating in the Stacks protocol by signing mined blocks.                            |
| **sBTC Signer**     | An entity that will sign sBTC operations and communicate with contracts on the chain to make that feasible. This entity has partial access to spending the sBTC UTXO.   |
| **sBTC Signer Set** | The set of all sBTC signers. Each is registered with the .sbtc contract and the transfer. These entities as a group have full democratic access to the sBTC UTXO.      |
| **sBTC Signer API** | An API exposed by the sBTC Signer that handles basic low-level commands.                                                                                                |

> | **Deposit API**     | A third party API that communicates with the sBTC Signers via the sBTC Signer API.                                                                                      |

Reviewer: ```suggestion
| **Deposit API**     | A third-party API that communicates with the sBTC Signers via the sBTC Signer API.                                                                                      |
```
fix grammar

The sBTC protocol not only addresses the limitations of Bitcoin's scripting system but also provides a secure and decentralized solution for utilizing Bitcoin in various applications.

## Design

While the first sBTC implementation is under development, the wrapped nature of the sBTC token means that any such system would have the following properties:

- sBTC is a SIP-10 token backed 1:1 by BTC.
- The sBTC peg wallet is maintained by the set of sBTC signers. These signers are responsible for the security and maintenance of the wallet, ensuring that sBTC is redeemable for BTC.
- Bitcoin can be converted into sBTC within 3 Bitcoin blocks, and sBTC can be converted into Bitcoin within 6 Bitcoin blocks. sBTC relies on the forking behavior guaranteed by [SIP-021](https://github.com/stacksgov/sips/blob/main/sips/sip-021/sip-021-nakamoto.md) in order to maintain the peg wallet correctly across forks.

## Specification

> Management of the sBTC peg wallet on the Bitcoin blockchain shall be managed by the proposed set of signers through a democratic process, involving the sBTC Signer Set rather than a single custodian. At launch, the sBTC protocol will be maintained by 15 independent entities that make up the sBTC Signer Set. The eligibility criteria to become an sBTC Signer is determined through the community governance process of ratifying this SIP.

Reviewer: ```suggestion
Management of the sBTC peg wallet on the Bitcoin blockchain shall be managed by the proposed set of signers through a democratic process, involving the sBTC Signer Set rather than a single custodian. At launch, the sBTC protocol will be maintained by 15 independent entities that make up the sBTC Signer Set. The eligibility criteria to become an sBTC Signer are determined through the community governance process of ratifying this SIP.
```
grammar

While up to 30% of the signers can be offline without a user impact on the functioning of the protocol, it becomes more critical for the rest of the signers to approve sBTC operations because operations necessarily still need to meet 70% of the original signing power. If more than 30% of signers become unavailable, no sBTC operations will be approved because it will be impossible to get 70% approval when less than 70% are online. An operation that isn’t approved will become reclaimable by the user after a timeout has elapsed. The timeout is specified by the user when preparing the deposit request and is measured in Bitcoin blocks.

### sBTC Signer Responsibilities

Overview of tasks the sBTC signers carry out:
- Accept requests to deposit BTC.
- Fulfill requests to withdraw BTC.
- Complete deposit and withdrawal requests in a timely manner.
- Maintain industry standard operational security around any hosts and private data (to include private keys).
- Move BTC to a new UTXO when private keys are rotated.

> - Signers must perform UTXO consolidate as it is deposited [1].

Reviewer: ```suggestion
- Signers must perform UTXO consolidation as it is deposited [1].
```
fix typo

- Signers must deduct transaction fees from users in order to fund BTC withdrawal transactions:
  - Ensure that the transaction fee is paid for (e.g., they deduct it from the user, and they set a minimum sBTC withdrawal amount).
  - Transaction fee must be estimated proportionally for the requested operation [2].
- Collectively, the signers must coordinate to calculate and advertise the fee parameters of the system:
  - A minimum sBTC peg-out.

>   - STX transaction fee for minting the sBTC. This fee is paid by the user and can be sponsored by a 3rd party, which is described in the [Auxiliary Features](#auxiliary-features) section of this document.

Reviewer: Auxiliary Features section which is referenced here is missing. Please add this section or remove the reference

Reviewer 2: removed

### sBTC Signer Eligibility Criteria

Signers will run the sBTC binary in addition to the core Stacks signer software and must meet certain criteria in order to facilitate the reliable functioning of the sBTC protocol at all times.

The following eligibility criteria will be used to identify the sBTC Signers:

> - Does the proposed sBTC signer have a demonstratable operating history which shows their experience and reliability in running blockchain services?

Reviewer: ```suggestion
- Does the proposed sBTC signer have a demonstrable operating history which shows their experience and reliability in running blockchain services?
```
fix typo

- Has the proposed sBTC signer participated in running a Stacks signer instance on Stacks 2.5 testnet or mainnet, and can they provide metrics showing this (ex: amount of stacks stacked over past several cycles)?
- Does the proposed sBTC signer agree to use reasonable efforts to maintain >99% uptime on the sBTC Signer?  
  *Note: This metric may be self-affirmed if independent verification is not possible, or confirmed by on-chain voting/stacking activity.*
- Does the proposed sBTC signer commit to a direct communication channel to be set up with the sBTC core engineers in order to respond to urgent updates within 24 hours?

> - Has the signer made contributions to Bitcoin or the Stacks network over the past year that demonstrate their commitment to the growth and success of the network? Examples include, but are not limited to: publishing independent research, marketing, co-authoring a SIP, submitting a Stacks pull request/issue, providing feedback on Stacks core development, or contributing to a Stacks working group.

Reviewer: ```suggestion
- Has the signer made contributions to Bitcoin or the Stacks network over the past year that demonstrate their commitment to the growth and success of the network? Examples include, but are not limited to: publishing independent research, marketing, co-authoring a SIP, submitting a Stacks pull request/issue, providing feedback on Stacks core development, or contributing to a Stacks Working Group.
```
'sBTC working group' should be capitalized as 'sBTC Working Group' for consistency

### [tBTC v2](https://whitepaper.io/document/691/tbtc-whitepaper)

tBTC is an open membership system, where the BTC is managed by a rotating set of randomly selected nodes which manage a threshold wallet. The system requires that 51-of-100 randomly selected wallet signers must collaborate to produce a proper signature.

### [RBTC](https://rootstock.io/static/a79b27d4889409602174df4710102056/RS-whitepaper.pdf)

rBTC, Rootstock’s (RSK) 2-way peg protocol, called “the Powpeg”, is a closed membership system. Peg operations settle to Bitcoin via merge mining on the RSK side-chain. Instead of collateralizing the system with a new token, peg operators are incentivized by earning a portion of transaction fees. PowPeg operators keep specialized hardware called PowHSMs active and connected to special types of Rootstock full nodes. Since the Bitcoin blockchain and the Rootstock sidechain are not entangled in a single blockchain or in a parent-child relation, peg-in and peg-out transactions require a high number of block confirmations. Peg-ins require 100 Bitcoin blocks, and peg-outs require 4000 Rootstock blocks.

## Activation

> sBTC is designed to activate on Stacks Nakamoto as defined in [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md). Therefore, this SIP is only meaningful when SIP-021 activates. The sBTC Working Group plans to observe at least 2-4 weeks of network behavior on Stacks Nakamoto to ensure a stable release. After this period, sBTC can be activated on the Stacks network without requiring a separate hard fork.

Reviewer: should Stacks Nakamoto be replaced with exact version number for consistency?

Reviewer 2: @andrerserrano

Author: Updated.

Solo stackers only can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.

#### For Non-Stackers:

Users with liquid STX can vote on proposals using the Ecosystem DAO. Liquid STX is the user’s balance, less any STX they have locked in the PoX stacking protocol, at the block height at which the voting started (preventing the same STX from being transferred between accounts and used to effectively double vote). This is referred to generally as "snapshot" voting.

For this SIP to pass, 66% of all liquid STX committed by voting must be in favor of the proposal. This precedent was set by [SIP-015](https://github.com/stacksgov/sips/blob/main/sips/sip-015/sip-015-network-upgrade.md).

The act of not voting is the act of siding with the outcome, whatever it may be. We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

## Appendix
[1] https://github.com/stacks-network/sbtc/issues/52

[2] https://github.com/stacks-network/sbtc/pull/186

> ### Specification

Reviewer: consider moving this to main Specification section, or renaming this to appropriate section name that could be referenced in main Specification

Reviewer 2: Not sure i follow the thinking here - structurally, this specification is part of the Appendix. 
are you asking to either move this section to the top level specification, or reference this appendix from the top-level?

@andrerserrano @aldur

Reviewer: either way is fine - moving to the top or if it better fits in appendix, then may be include references within top level specification

Reviewer 2: i think i see what you're getting at - so in the case where this appendix specification remains where it is, you'd like to see the top-level specification annotate to the bottom-level specification?

Author: @jiga We intentionally did not include this in the Spec because how deposits and withdrawals work are outside of the scope for this SIP. I added a note in the Spec section for the reader find more detail in the appendix.  https://github.com/stacksgov/sips/pull/186/commits/3aece3aad3fab822bf7ef7056a07990f02b73c05

Reviewer: thanks @andrerserrano this looks good now

#### Deposits

The main steps of the sBTC Deposit flow will be as follows:

1. **Deposit request:** A bitcoin holder creates a transaction on Bitcoin.
2. The deposit transaction contains a UTXO (deposit UTXO) spendable by sBTC Signers, with an OP_DROP payload.
3. The payload contains the recipient address of the sBTC among other relevant info for the deposit.
   - The relevant info could contain a fee suggestion or max_fee.
4. **Proof of deposit:** The bitcoin holder submits a proof of deposit on Stacks by invoking the Deposit API.

> 5. **Deposit accept:** 

Reviewer: missing details here

Reviewer 2: @andrerserrano @aldur

Author: Can you elaborate - which details would you like to see? The SIP is on the signer criteria, not on the technical details of the deposit flow. We aimed to provide a high level overview of the system without going into too many tech details.

Reviewer 2: `5. **Deposit accept:** `

has no definition like the other items do

Author: Got it. That was a formatting issue, these were meant to be sub-bullets. I've updated this - thanks!

# SIP-028: sBTC Signer Criteria

**SIP Number:** 028  
**Title:** Signer Criteria for sBTC, A Decentralized and Programmable Asset Backed 1:1 with BTC  
**Consideration:** Technical, Economics  
**Type:** Standard  
**Status:** Draft  
**Authors:**  
- Andre Serrano ([andre@bitcoinl2labs.com](mailto:andre@bitcoinl2labs.com))  
- Ashton Stephens ([ashton@trustmachines.co](mailto:ashton@trustmachines.co))  
- Joey Yandle ([joey@trustmachines.co](mailto:joey@trustmachines.co))  
- Mårten Blankfors ([marten@trustmachines.co](mailto:marten@trustmachines.co))  
- Jesus Najera ([jesus@stratalabs.xyz](mailto:jesus@stratalabs.xyz))  
- Jude Nelson ([jude@stacks.org](mailto:jude@stacks.org))  
- Friedger Müffke ([friedger@ryder.id](mailto:friedger@ryder.id))  
- Tycho Onnasch ([tycho@zestprotocol.com](mailto:tycho@zestprotocol.com))  
- Daniel Jordon ([daniel@trustmachines.co](mailto:daniel@trustmachines.co))

## Abstract

> This SIP takes the position that Stacks can play a key role in offering a rich programming environment for Bitcoin with low-latency transactions. This would be achieved with a new wrapped Bitcoin asset, called sBTC, which would be implemented on Stacks 3.0 and later as a SIP-010 token. Stacks today offers a smart contract runtime for Stacks-hosted assets, and the forthcoming Stacks [3.0 release](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md) provides lower transaction latency than Bitcoin for Stacks transactions. By providing a robust BTC-wrapping mechanism based on [threshold signatures](https://eprint.iacr.org/2020/852.pdf), users would be able to lock their real BTC on the Bitcoin chain, instantiate an equal amount of sBTC tokens on Stacks, use these sBTC tokens on Stacks, and eventually redeem them for real BTC at 1:1 parity, minus the cost of the relevant blockchain transaction fees.

Reviewer 3: Nit: Shorten `This SIP takes the position that Stacks can play a key role in offering a rich programming environment for Bitcoin with low-latency transactions. This would be achieved with a new wrapped Bitcoin asset...` to `This SIP proposes a new wrapped Bitcoin asset...`.

Reviewer 2: d5b1ac0750e2d5c5690f87a8f5473ad755019545

> This SIP takes the position that Stacks can play a key role in offering a rich programming environment for Bitcoin with low-latency transactions. This would be achieved with a new wrapped Bitcoin asset, called sBTC, which would be implemented on Stacks 3.0 and later as a SIP-010 token. Stacks today offers a smart contract runtime for Stacks-hosted assets, and the forthcoming Stacks [3.0 release](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md) provides lower transaction latency than Bitcoin for Stacks transactions. By providing a robust BTC-wrapping mechanism based on [threshold signatures](https://eprint.iacr.org/2020/852.pdf), users would be able to lock their real BTC on the Bitcoin chain, instantiate an equal amount of sBTC tokens on Stacks, use these sBTC tokens on Stacks, and eventually redeem them for real BTC at 1:1 parity, minus the cost of the relevant blockchain transaction fees.

Reviewer 2: ```suggestion
This SIP takes the position that Stacks can play a key role in offering a rich programming environment for Bitcoin with low-latency transactions. This would be achieved with a new wrapped Bitcoin asset, called sBTC, which would be implemented on Stacks 3.0 and later as a SIP-010 token. Stacks today offers a smart contract runtime for Stacks-hosted assets, and the forthcoming Stacks [3.0 release](https://github.com/stacksgov/sips/blob/main/sips/sip-021/sip-021-nakamoto.md) provides lower transaction latency than Bitcoin for Stacks transactions. By providing a robust BTC-wrapping mechanism based on [threshold signatures](https://eprint.iacr.org/2020/852.pdf), users would be able to lock their real BTC on the Bitcoin chain, instantiate an equal amount of sBTC tokens on Stacks, use these sBTC tokens on Stacks, and eventually redeem them for real BTC at 1:1 parity, minus the cost of the relevant blockchain transaction fees.
```

This SIP outlines but does not describe in technical detail the workings of the first sBTC system. A separate SIP will be written to do so if this SIP successfully activates.

## Introduction

### Glossary

| Term                | Definition                                                                                                                                                            |
|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **SIP-10 Token**    | A token on the Stacks blockchain that adheres to the fungible token standards outlined in [SIP-10](https://github.com/stacksgov/sips/blob/main/sips/sip-010/sip-010-fungible-token-standard.md).                                                                       |
| **sBTC**            | A SIP-10 token on the Stacks Blockchain that can be turned back into BTC on the Bitcoin Blockchain. 1 sBTC is equivalent to 1 BTC on the Bitcoin Blockchain.           |

> | **sBTC operation**  | An operation that initiates some action from the sBTC protocol.                                                                                                        |

Reviewer 3: Nit: This is a recursive definition.  What's an "operation?"

Reviewer 2: alt: `An on-chain (function/tx) that initiates some action from the sBTC protocol`

Reviewer 2: d5b1ac0750e2d5c5690f87a8f5473ad755019545

Doing this in an open-membership peer-to-peer setting has been shown to be a difficult task (given the size and complexity of systems like Lightning, Taro, Bisq, and BitVM), which imposes a high barrier to entry for building decentralized applications on Bitcoin. Our key insight is that existing applications built on Bitcoin have built up most of the workings of an L2 blockchain like Stacks, but have done so implicitly within their interior components. This SIP makes this design choice explicit: the act of building non-trivial applications on Bitcoin is the act of building on a Bitcoin L2, and therefore the act of providing a rich programming environment for BTC is the act of implementing a wrapped BTC asset (sBTC) on a Bitcoin L2 with smart contracts. Indeed, this has been realized already with systems like Rootstock's RBTC.

## Proposed Solution

> sBTC aims to mitigate Bitcoin’s limitations by combining the capability of the Stacks Blockchain with the reliability and security of Bitcoin. By enabling the secure movement of BTC in and out of the Stacks Blockchain via the sBTC protocol, users can interact with their BTC on Stacks using Clarity smart contracts and fast block times. The protocol is “secure” in that it is operated by a decentralized signer network, removing the risk of a single point of failure and trust in a single custodian. Users can deposit BTC into the protocol, seamlessly transact using sBTC on the Stacks blockchain, and have the freedom to redeem sBTC tokens for the underlying BTC at any time.

Reviewer 2: nit: `fast` block times is nebulous. i would either remove that part, or modify like "faster block times than Bitcoin provides"

### Programmability

[Clarity](https://docs.stacks.co/clarity/overview) is the smart contract language on Stacks, which allows developers to encode essential business logic on a blockchain. Using smart contracts, developers can build more expressive decentralized applications that interact with sBTC, such as DeFi protocols, stablecoins, payments, and many others.

### Fast Blocks

> The Stacks Nakamoto Upgrade, proposed in [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md#proposed-solution), enables fast blocks where user-submitted transactions will now take on the order of seconds, instead of tens of minutes. Thus, sBTC on Stacks Nakamoto will offer an improvement to Bitcoin’s current transaction times.

Reviewer 2: ```suggestion
The Stacks Nakamoto Upgrade, proposed in [SIP-021](https://github.com/stacksgov/sips/blob/main/sips/sip-021/sip-021-nakamoto.md#proposed-solution), enables fast blocks where user-submitted transactions will now take on the order of seconds, instead of tens of minutes. Thus, sBTC on Stacks Nakamoto will offer an improvement to Bitcoin’s current transaction times.
```

> The sBTC protocol not only addresses the limitations of the Bitcoin scripting system but also provides a secure and decentralized solution for utilizing Bitcoin in various applications.

Reviewer 2: ```suggestion
The sBTC protocol not only addresses the limitations of Bitcoin's scripting system, but also provides a secure and decentralized solution for utilizing Bitcoin in various applications.
```

## Design

While the first sBTC implementation is under development, the wrapped nature of the sBTC token means that any such system would have the following properties:

- sBTC is a SIP-10 token backed 1:1 by BTC.
- The sBTC peg wallet is maintained by the set of sBTC signers. These signers are responsible for the security and maintenance of the wallet, ensuring that sBTC is redeemable for BTC.

> - Bitcoin can be converted into sBTC within 3 Bitcoin blocks; and sBTC can be converted into Bitcoin within 6 Bitcoin blocks. sBTC relies on the forking behavior guaranteed by [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md) in order to maintain the peg wallet correctly across forks.

Reviewer 3: Nit: the use of the semicolon is grammatically incorrect.  Use a comma please.

Reviewer 3: > sBTC relies on the forking behavior guaranteed by [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md) in order to maintain the peg wallet correctly across forks.

I'm a little confused by this.  Can you specifically state the forking rules you're relying on?  Asking because while the first production release of Nakamoto guarantees that each Bitcoin fork has at most one Stacks fork after two confirmations, it doesn't guarantee anything about the confirmation of transactions which land on a Stacks fork whose Bitcoin fork gets orphaned.  The SIP states that this will be addressed after the release.  I just want to confirm that sBTC does not depend on this particular behavior?  Because if it does, then this SIP would be blocked on that, and it should be stated up-front.

Reviewer 2: @aldur

Author: This does not refer to transaction replay, which is outside of the scope of the SIP. It is referring to the fact that Stacks blockchain will no longer fork on its own, it will always follow Bitcoin forks.

## Specification

> Management of the sBTC peg wallet on the Bitcoin blockchain is decentralized, involving the sBTC Signer Set rather than a single custodian. At launch, the sBTC protocol will be maintained by 15 independent entities that make up the sBTC Signer Set. The eligibility criteria to become an sBTC Signer is determined through the community governance process of ratifying this SIP.

Reviewer 3: I think the word "decentralize" is being twisted here to imply something that you don't mean.  The signer set is going to be chosen by this SIP, but the signer set is fundamentally closed-membership once this SIP is ratified, right?  Specifically, the signer set is going to be chosen by the sBTC working group, as mentioned later?

Reviewer 2: i think agree with the author here - decentralized doesn't implicitly imply open membership, even though that may be how it's mostly understood today. 
a system may have closed membership and still be decentralized (assuming none of the closed members can manipulate a system)  - unless there's a definition i'm missing?

Reviewer 3: I mean, the bigger problem is that "decentralized" is a very loaded buzzword with no technically-meaningful definition, so it's just as hard to promote its use as it is to discourage it.  However, the "vibes" I get is that "decentralized" implies "no distinguished authority decides how this works," which is _not_ true here.  This SIP is a distinguished authority that declares what the signer set shall be.

Reviewer 2: that's a fair point. borrowing some terms from elsewhere in this document, what about a change like this?
```
Management of the sBTC peg wallet on the Bitcoin blockchain shall be managed by the proposed set of signers through a democratic process...
```

Reviewer 2: d5b1ac0750e2d5c5690f87a8f5473ad755019545

sBTC Signers are responsible for accepting or rejecting all sBTC deposit and withdrawal operations submitted to the network. For a transaction to be fulfilled, at least 70% of the signers need to approve the transaction. This means that the liveness and reliability of the signers is crucial to the success of the protocol. The system is live ("resilient") if at least 70% of the sBTC Signer voting power are online and honest. Then (and only then), deposits and withdrawals happen in a timely manner. The system is safe ("trustworthy") if at least 30% of the sBTC Signer voting power is honest. Then, no theft of funds can occur.

> Throughout this release, sBTC will have an unchangeable and distinct signer set that will not explicitly be part of Stacks consensus. As a result, this release can activate without a hard fork. Each unique signer receives exactly one vote. Given there are 15 unique signers that comprise the system, it would require at least 11 out of 15 signatures for an sBTC operation to be fulfilled.

Reviewer 3: > As a result, this release can activate without a hard fork.

Instead of talking about forks, it would be more accurate to point out that sBTC is an _application_ on top of Stacks, and doesn't need any special treatment.

Reviewer 2: removed in 91ce0b64bd03c643983dba7a587ae08be57d1f16

> While up to 30% of the signers can be offline without a user impact on the functioning of the protocol, it becomes more critical for the rest of the signers to approve sBTC operations because operations necessarily still need to meet 70% of the original signing power. If more than 30% of signers become unavailable, no sBTC operations will be approved because it will be impossible to get 70% approval when less than 70% are online. An operation that isn’t approved will become spendable by the user after a predetermined timeout has elapsed, measured in Bitcoin blocks.

Reviewer 2: can you define what this "predetermined timeout" shall be?

Reviewer 4: Address in 671ad9d

### sBTC Signer Responsibilities

Overview of tasks the sBTC signers carry out:
    - They react to requests to deposit BTC.
    - They react to requests to withdraw BTC.
    - They stay online often enough that deposits and withdrawals happen in a timely manner.
    - They have to keep their signer hosts secure, so their keys don't get stolen.

>     - They have to proactively re-key every so often, which means moving the BTC to a new UTXO.

Reviewer 3: How often do they need to re-key?

Reviewer 2: i'm not comfortable with the wording in this list - it reads like a list of what a signer *shall* do, but without mention of how it's measured or what consequences there are for failing to meet them. 

I think a slight rewording could work though, ex:
```
- Accept requests to deposit BTC.
- Accept requests to withdraw BTC.
- Complete deposit and withdrawal requests in a timely manner. 
- Maintain industry standard operational security around private keys.
  - Move BTC to a new UTXO when keys are rotated. 
```

I've removed the "proactively" line here, since i think industry standard opsec would imply the key rotation so the relevant responsibility is that the btc is moved as part of that implied key rotation. 

the one question that may come up though is what does `timely manner` mean here? it would be helpful to put a specific number here, i.e. 6 bitcoin blocks

Reviewer 2: d5b1ac0750e2d5c5690f87a8f5473ad755019545

i've removed the implicit line around rotating keys, with it being implied by following standard opsec. i think this addresses your comment here

> - Signers must consolidate BTC as it is deposited.

Reviewer 3: Nit: instead of linking to issues, can you put a summary of the mechanism in an appendix?  SIPs should be as complete as can be, so that as long as someone has a copy of the SIP (and only the SIP), they can develop a comprehensive understanding of the proposal without needing to do additional research.

Reviewer 2: d5b1ac0750e2d5c5690f87a8f5473ad755019545

    - The process for UTXO consolidation of the BTC peg wallet is described in this [issue](https://github.com/stacks-network/sbtc/issues/52).
- Signers must deduct transaction fees from users in order to fund BTC withdrawal transactions.
    - They must ensure that the transaction fee is paid for (e.g., they deduct it from the user, and they set a minimum sBTC withdrawal amount).
    - The process for handling transaction fee estimates is described in this [issue](https://github.com/stacks-network/sbtc/pull/186).
- Signers must coordinate to set and advertise the fee parameters of the system.
    - They must decide a minimum sBTC peg-out.
    - They must decide an STX transaction fee for minting the sBTC. This fee is paid by the user and can be sponsored by a 3rd party, which is described in the “Auxiliary Features” section of this document.

### sBTC Signer Eligibility Criteria

> Signers will run the sBTC binary in addition to the core Stacks signer software and must meet certain criteria in order to facilitate the reliable functioning of the sBTC protocol at all times.

Reviewer 4: Why is it required to run a core Stacks signer?

Reviewer 3: I'm also confused by this.  I can see a reason for sBTC signers to run Stacks nodes, but why signers?  Are sBTC signers required to first be block signers in Nakamoto?

Reviewer 2: I read this as a proposed requirement for an initial sBTC signer set is that they also run a stacks node signer. 
there's no technical reason this needs to be true, but for the eligibility of the first set of chosen signers - this is an imposed requirement. is that what you're suggesting @andrerserrano ?

Reviewer 2: Adding on here - i think the language here needs a little update to remove pronouns. 
ex: 
`They must decide a minimum sBTC peg-out.` -> `A minimum sBTC peg-out must be decided`

keeping these responsibilities agnostic reads a bit better than adding a pronoun "they" (etc) to each item in the list.

Reviewer 2: addressed the pronouns in the list. 
@aldur how do you feel about the explantation around the sbtc signer also running a stacks signer?

Author: Yes, running a Stacks signer is not a technical requirement, but it shows that they have experience with the codebase and provides more data to evaluate them by.

Reviewer 3: I had to go look at the sBTC signer code to determine this, but I think it's relevant to call out that the sBTC signer is a Stacks event observer, meaning that the experience of running a Stacks node signer directly translates to running an sBTC signer (this isn't obvious from the text).

They don't have to be a Stacker on mainnet, right?  It suffices that they run a Stacks signer as part of mock-signing or on testnet as well, right?

Author: That's correct. Good call out, I will add that text to the SIP.

The following eligibility criteria will be used to identify the sBTC Signers:

> - **Company Standing:**  

Reviewer 3: Does this mean that only _companies_ can be sBTC signers?  Do sBTC signers need to be legally incorporated, and if so, why?

  Does the company have an operating history to demonstrate their experience and reliability?

> - **Stacks Participation:**  

Reviewer 3: Why is this necessary?  And, do you have a means of proving that they have done so?

Reviewer 2: remove in 321d118f5dd48f1f526bd6d9cfd89497878ac98b

  Has the Signer participated in running a signer node on Stacks 2.5 testnet or mainnet?
- **Network Uptime:**  

>   Does the Signer agree to use reasonable efforts to maintain >99% uptime on the sBTC Signer?  

Reviewer 3: How will we know that they do?

Reviewer 2: 321d118f5dd48f1f526bd6d9cfd89497878ac98b

>   *Note: This metric will be self-affirmed by the company.*

Reviewer 3: Can you do something a little bit stronger?  This sounds like the signer can just say "trust me bro" and pass this criterion?

Reviewer 2: what about :
```
*Note: This metric may be self-affirmed by the company if independent verification is not possible.*
```
i.e. for a nat'ed instance, it's impossible to check for open ports or much of anything really

Reviewer 2: 321d118f5dd48f1f526bd6d9cfd89497878ac98b

- **Communication & Availability:**  

>   Does the signer have a direct communication channel set up with sBTC core engineers to be able to respond to urgent updates within 24 hours? (e.g., Slack, Telegram, Signal)

Reviewer 3: Is there a way for them to demonstrate their availability?  As in, is there a way for members of the public to confirm that they're present in these channels?

Also, why are there more than one channel?  Don't we want all the signers to be using the same channel, so there's a single well-known way to get ahold of them if something breaks?

Reviewer 2: i beleive this is addressed with changes in 321d118f5dd48f1f526bd6d9cfd89497878ac98b

- **Ecosystem Alignment:**  

>   Has the signer made contributions to Bitcoin or the Stacks network over the past year that demonstrate their commitment to the growth and success of the network? Examples include, but are not limited to: publishing independent research, marketing, co-authoring a SIP, submitting a Stacks pull request, providing feedback on Stacks core development, or contributing to a Stacks working group.

Reviewer 2: yes, this will be provided and is in process already.

The criteria described above will be used to identify sBTC Signers that are able to meet some or all of the responsibilities described in the previous section.

### Selection Process

> The sBTC Signer Set will be finalized from the list of eligible Signers, based on the above criteria, by the [sBTC working group](https://github.com/orgs/stacks-network/discussions/469). This process is to ensure that the sBTC Working Group can adjust to any changes in the sBTC Signer Set quickly and without the overhead of an additional community vote. For example, if an sBTC Signer is no longer available to complete their responsibilities, it is imperative for the liveness of the system that the sBTC Working Group is able to implement a suitable replacement based on the above criteria.

Reviewer 3: > This process is to ensure that the sBTC Working Group can adjust to any changes in the sBTC Signer Set quickly and without the overhead of an additional community vote.

I think this may be a case of you "saying the quiet part out loud."  I can understand the desire to avoid the time costs of a long-running vote, but surely there must be some means for community oversight of the signer set?

Reviewer 3: I think once you have an initial signing set, then you can have future SIPs for changing it.  Unlike this SIP, I think there's room for SIPs that only change the signing set, which can have shorter voting windows and only require participation from sBTC holders (since they'd be the only stakeholders).

Reviewer 2: i believe this is addressed by 09944a5ecc79ca0852b3ef2cea9e2b7c407d1644

> ## Related Work

Reviewer 3: Do you want to also cite some wrapped BTC assets in the Stacks ecosystem?

Reviewer 2: Since existing assets weren't used in teh design of sBTC, we felt it was not necessary here.

### [tBTC v2](https://whitepaper.io/document/691/tbtc-whitepaper)

tBTC is an open membership system, where the BTC is managed by a rotating set of randomly selected nodes which manage a threshold wallet. The system requires that 51-of-100 randomly selected wallet signers must collaborate to produce a proper signature.

### [RBTC](https://rootstock.io/static/a79b27d4889409602174df4710102056/RS-whitepaper.pdf)

rBTC, Rootstock’s (RSK) 2-way peg protocol, called “the Powpeg,” is a closed membership system. Peg operations settle to Bitcoin via merge mining on the RSK side-chain. Instead of collateralizing the system with a new token, peg operators are incentivized by earning a portion of transaction fees. PowPeg operators keep specialized hardware called PowHSMs active and connected to special types of Rootstock full nodes. Since the Bitcoin blockchain and the Rootstock sidechain are not entangled in a single blockchain or in a parent-child relation, peg-in and peg-out transactions require a high number of block confirmations. Peg-ins require 100 Bitcoin blocks, and peg-outs require 4000 Rootstock blocks.

> 

Reviewer 3: As part of the related work section, can you summarize the important ways by which sBTC differs?  This helps the reader understand why sBTC is necessary -- what niche is it filling that these other systems do not fill, and cannot be trivially adapted to fill?

Reviewer 2: 0abe397308d1a59f71501bae6be32d0681abccd8

Reviewer 3: I'm not sure what "Bitcoin-Aligned Solution" means or how to parse its definition.  Sounds too much like marketing (ambiguous, unquantifiable, and unfalsifiable).

Maybe just drop it?

Author: Done.

## Activation

> sBTC is designed to activate on Stacks Nakamoto as defined in [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md). Therefore, this SIP is only meaningful when SIP-021 activates. The sBTC Working Group plans to observe at least 2-4 weeks of network behavior on Stacks Nakamoto to ensure a stable release. After this period, sBTC can be activated on the Stacks network without requiring a separate hard fork.

Reviewer 3: I think a missing but necessary activation criterion will be that the SIP authors put forth the list of signers, and their reasons why they have met the above criteria better than any other candidate (it would also be a good idea to list all of the candidates, even if they did not get selected).

Reviewer 3: If the sBTC working group can get together the above data, then at least the SC and user community will not be devolving their authority to advance and ratify SIPs (a [strong point of contention](https://github.com/stacksgov/sips/pull/186#discussion_r1688478800) I had earlier).  The 15 signers would themselves be part of the SIP, as well as the rationale for including them over others.  Then, the sBTC working group and its signer set can be held accountable by the user community as part of the activation process.

This also has the nice consequence of avoiding the need for the SC to independently vet the signer set according to the SIP's critiera, which would be very difficult to do as-written due to the ambiguity of the criteria.  If there is any contention between the SC and user community with the sBTC working group over the signer set, then we can visit it before activation by way of simply re-considering the ultimate signer list instead of re-considering the means by which they were discovered and ranked.

### Process of Activation

Users can vote to approve this SIP with either their locked/stacked STX or with unlocked/liquid STX, or both. The criteria for the stacker and non-stacker voting is as follows.

#### For Stackers:

In order for this SIP to activate, the following criteria must be met by the set of Stacked STX:

- At least 80 million Stacked STX must vote at all to activate this SIP.
- Of the Stacked STX that vote, at least 66% of them must vote "yes."

The voting addresses will be:

> - **Bitcoin Yes Address:** `3Jq9UT81fnT2t24XjNVY7wijpsSmNSivbK`  

Reviewer 3: Can you explain the origins of these addresses?

In prior SIPs, we'd encode a string like "yes-to-3.0" or "no-to-3.0" as base58check strings, so it's unambiguous that these addresses aren't being used to e.g. line the pockets of the SIP authors, or e.g. are being used by some other protocol which would skew or obfuscate the vote.

Reviewer 4: Address in f09f236, thank you!

Reviewer 2: lgtm

For this SIP to pass, 66% of all liquid STX committed by voting must be in favor of the proposal. This precedent was set by [SIP-015](https://github.com/stacksgov/sips/blob/feat/sip-015/sips/sip-015/sip-015-network-upgrade.md).

The act of not voting is the act of siding with the outcome, whatever it may be. We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

## Appendix

### Specification

#### Deposits

The main steps of the sBTC Deposit flow will be as follows:

1. **Deposit request:** A bitcoin holder creates a transaction on Bitcoin.
2. The deposit transaction contains a UTXO (deposit UTXO) spendable by sBTC Signers, with an OP_DROP payload.
3. The payload contains the recipient address of the sBTC among other relevant info for the deposit.
   - The relevant info could contain a fee suggestion or max_fee.

> 4. **Proof of deposit:** The bitcoin holder submits a proof of deposit on Stacks by invoking the Signer binary API.

Reviewer 4: ```suggestion
4. **Proof of deposit:** The bitcoin holder submits a proof of deposit on Stacks by invoking the sBTC Signer binary API.
```

#### Withdrawals (Redeeming sBTC)

The main steps of the sBTC withdrawal flow are as follows:

1. **Withdrawal request:** An sBTC holder calls the `withdraw-request` function in the `.sbtc` contract.
   - This transfers the requested amount of sBTC to the `.sbtc` contract & mints the user a non-transferable locked-sBTC as a placeholder.
2. **Withdrawal accept:** If accepted, the following happens:
   - The signers create a transaction on Bitcoin which returns the requested amount to the designated address.
   - Once the Bitcoin transaction is confirmed, the signers make a smart contract call to one of the `.sbtc` contracts to mark the transaction as fulfilled.
   - If successful, the resulting Stacks transaction will record the withdrawal request as complete & will accordingly burn the user’s locked-sBTC.
3. **Withdrawal reject:** If instead the request is rejected, the sBTC signers will call the `withdraw-reject` function in the `.sbtc` smart contract. This function does the following:
   - Returns the sBTC to the holder.
   - Records the signer votes.

### Auxiliary Features

Auxiliary features of the sBTC protocol are described below.

#### Stacks Transaction Fee Sponsorship

> sBTC will include the option to have sBTC transactions on Stacks be sponsored in return for some sBTC. Using the approach suggested in this [issue](https://github.com/stacks-network/stacks-core/issues/4235), sBTC users will be able to pay for their transaction fees in sBTC with support from an existing STX holder, provided the wallet supports it. The proposed solution introduces atomic transaction bundles on Stacks, which enable sBTC payments to sponsors for covering STX transaction fees. STX is maintained as the sole gas token, but the user only has to interact with sBTC.

Reviewer 3: Can you summarize the issue here, and then link to it?  What about transaction bundles, specifically, is necessary to get to using only sBTC to pay for transactions?

Also, this subsection should have the bottom line up front, instead of describing the means to the end.  Can you call it something like "Paying Transaction Fees in sBTC" or something?

Reviewer 2: I feel like this section isn't even needed for this SIP. There is an open issue regarding the method to accomplish this, a forum post, and an implied future SIP to define implementation. 

I would argue it shouldn't be here, since it's an implementation detail - either a future sip is ratified that codifies it, or the specific sbtc contracts implement this on their own.

Reviewer 2: removed in 321d118f5dd48f1f526bd6d9cfd89497878ac98b

#### Signer Key Rotation

> Mechanisms are provided for the scenario where a signer wants to rotate their key. For this to happen, signers must coordinate offline and vote on-chain on the new signer set (aka set of keys). Once the new signer set is determined, the signers conduct a wallet handoff and re-execute the key generation event.

Reviewer 3: What mechanisms, specifically?

Reviewer 2: removed in 321d118f5dd48f1f526bd6d9cfd89497878ac98b

## Specification

Management of the sBTC peg wallet on the Bitcoin blockchain shall be managed by the proposed set of signers through a democratic process, involving the sBTC Signer Set rather than a single custodian. At launch, the sBTC protocol will be maintained by 15 independent entities that make up the sBTC Signer Set. The eligibility criteria to become an sBTC Signer are determined through the community governance process of ratifying this SIP.

sBTC Signers are responsible for accepting or rejecting all sBTC deposit and withdrawal operations submitted to the network. For a transaction to be fulfilled, at least 70% of the signers need to approve the transaction. This means that the liveness and reliability of the signers is crucial to the success of the protocol. The system is live ("resilient") if at least 70% of the sBTC Signer voting power are online and honest. Then (and only then), deposits and withdrawals happen in a timely manner. The system is safe ("trustworthy") if at least 30% of the sBTC Signer voting power is honest. Then, no theft of funds can occur. Additionally, more details on sBTC deposit and withdrawals are included in the appendix of this SIP. 

> Throughout this release, sBTC will have an unchangeable and distinct signer set that will not explicitly be part of Stacks consensus. Given that there will be 15 unique signers that shall comprise the system, and each unique signer is allocated exactly one vote, it would require at least 11 out of 15 signatures for an sBTC operation to be fulfilled.

Reviewer 3: I'm not sure this is true in the sBTC code.  It seems that signers can "vote in" other signers and "vote out" others if 11 of 15 signers vote to transfer the BTC to a new Bitcoin address, such that signature data from a _new, different_ signer set is required to spend it.

I don't have an opinion either way as to whether or not this is acceptable (the community will decide), but the SIP should be consistent with the implementation.

Reviewer 3: Like, even if this isn't in the sBTC code as an explicit feature, the fact that sBTC supports private key rotation (mentioned below) is equivalent to sBTC supporting changing the signer set.  So unless this SIP mandates that the BTC deposit address is fixed forever, sBTC supports rotating signers.

Reviewer 3: Ah, I see you have a subsection about upgrading the sBTC signer set below.  In that case, I think this paragraph needs to be revised.  The signer set is not "unchangeable".

Also, it is _not_ part of consensus.  No "explicitly" about it.

Author: Agreed. I've updated the SIP text to clarify that the sBTC signer set _is_  changeable and _not_ part of Stacks consensus.

> While up to 30% of the signers can be offline without a user impact on the functioning of the protocol, it becomes more critical for the rest of the signers to approve sBTC operations because operations necessarily still need to meet 70% of the original signing power. If more than 30% of signers become unavailable, no sBTC operations will be approved because it will be impossible to get 70% approval when less than 70% are online. An operation that isn’t approved will become reclaimable by the user after a timeout has elapsed. The timeout is specified by the user when preparing the deposit request and is measured in Bitcoin blocks.

Reviewer 3: Is there a plan for recovery if more than six signers go permanently offline? Is the Bitcoin stuck forever in that case, or is there some other spending condition that lets it be reclaimed after e.g. a timeout?  Either way, the SIP should state what happens in this case.

Author: cc: @aldur @AshtonStephens could we develop a spend condition that lets it be reclaimed in this scenario? If not, I'll specify it in the SIP.

Reviewer 3: Minor word smithing.  Suggest replacing this:

`An operation that isn’t approved will become reclaimable by the user after a timeout has elapsed. The timeout is specified by the user when preparing the deposit request and is measured in Bitcoin blocks.`

With this:

`To protect users from a liveness failure during deposit, a deposit UTXO shall be made satisfiable by one of two spending conditions: (1) the signer set spends the UTXO, or (2) the user spends the UTXO after a fixed number of Bitcoin blocks have passed.  Then, if there is an indefinite liveness failure, users will be able to reclaim their in-flight BTC.`

If you have a script template, you could (at your discretion) include it in an appendix.

Author: Added Jude's suggestions on liveness failure and included the format in the appendix. 3266c2a13e3c8ba059c8253e68d84ccaa96fd491

### sBTC Signer Responsibilities

> Overview of tasks the sBTC signers carry out:

Reviewer 3: Nit: Can these at least be complete sentences?  Also, I think these responsibilities can be further factored into things that the sBTC protocol makes signers do, and things that signers ought to be doing to run sBTC.

For example, sBTC signers _ought to_ be maintaining industry-standard opsec, performing UTXO consolidation, deducting transaction fees, and advertising the fee parameters.  These things aren't part of the sBTC protocol.

Author: Updated in 0b5a719

- Collectively, the signers must coordinate to calculate and advertise the fee parameters of the system:
  - A minimum sBTC peg-out.
  - STX transaction fee for minting the sBTC. This fee is paid by the user and can be sponsored by a 3rd party.

### sBTC Signer Eligibility Criteria

Signers will run the sBTC binary in addition to the core Stacks signer software and must meet certain criteria in order to facilitate the reliable functioning of the sBTC protocol at all times.

The following eligibility criteria will be used to identify the sBTC Signers:

- Does the proposed sBTC signer have a demonstrable operating history which shows their experience and reliability in running blockchain services?
- Has the proposed sBTC signer participated in running a Stacks signer instance on Stacks 2.5 testnet or mainnet, and can they provide metrics showing this (ex: amount of stacks stacked over past several cycles)?
  * Note: The sBTC signer is a Stacks event observer, meaning that the experience of running a Stacks node signer directly translates to running an sBTC signer.
- Does the proposed sBTC signer agree to use reasonable efforts to maintain >99% uptime on the sBTC Signer? 

>   * Note: This metric may be self-affirmed if independent verification is not possible, or confirmed by on-chain voting/stacking activity.*

Reviewer 3: Typo: trailing asterisk

- Does the proposed sBTC signer commit to a direct communication channel to be set up with the sBTC core engineers in order to respond to urgent updates within 24 hours?
- Has the signer made contributions to Bitcoin or the Stacks network over the past year that demonstrate their commitment to the growth and success of the network? Examples include, but are not limited to: publishing independent research, marketing, co-authoring a SIP, submitting a Stacks pull request/issue, providing feedback on Stacks core development, or contributing to a Stacks Working Group.

> The criteria described above will be used to identify sBTC Signers that are able to meet some or all of the responsibilities described in the previous section.

Reviewer 3: Might be worth mentioning in this section that the sBTC working group will be performing the vetting, instead of waiting until the `Activation` section to mention it.

Reviewer 3: Also, are the sBTC developers part of the sBTC working group?  Is there a list of personnel somewhere who comprise the sBTC working group?

Author: Completed 3d680c1a2c2061a1c7ac8aa7622298e19b488621

### Selection Process
The initial sBTC Signer Set will be finalized from the list of eligible Signers, based on the above criteria. The results will be published as a discussion in the [sBTC Working Group](https://github.com/stacks-network/sbtc/discussions/624).

The selection process is as follows:
1. **Nomination Phase**: Open a call for nominations within the community.
2. **Evaluation & Community Feedback**: The proposed signer set will be published to provide transparency.
3. **SIP Vote**: The community will vote on the signer criteria and proposed signer set.

> 4. **Final Selection**: The final list of signers will be selected based on community feedback and successful completion of SIP-028.

Reviewer 3: I think it needs to be explicitly stated that **the signer set the community votes on will be the signer set for sBTC.** No further additions or removals will be done once the SIP vote begins.

Reviewer 3: Sorry, I should have been more clear.

This text is too ambiguous because it leaves open the possibility that the signer set could be changed (i.e. the set is "based on" community feedback, as opposed to "wholly determined by.").

How about this instead?

"4. **Final Selection**:  If SIP-028 is ratified, then the proposed signer set voted upon in step 3 shall be the initial signing set for sBTC.  If the signer set changes during the vote, such as by the withdrawal of one or more candidates, then the vote will be restarted in a subsequent reward cycle (to be determined if this comes to pass)."

> ### Updating The sBTC Signer Set

Reviewer 3: This directly contradicts the third paragraph of the `Specification` section.

Author: Updated the Specification section to correct this.

In the event that the sBTC Signer Set needs to be updated (for example, if a signer is no longer available to complete their responsibilities) sBTC Signers can perform a threshold vote to agree on the updated set. This process will also be performed if a signer needs to rotate their cryptographic keys.

## Related Work

### [WBTC](https://wbtc.network/assets/wrapped-tokens-whitepaper.pdf)

WBTC is a closed membership system. It is made up of 50+ merchants and custodians with keys to the WBTC multisig contract on Ethereum. WBTC deposits and withdrawals can only be performed by the authorized merchants, and end users purchase WBTC directly from the merchants. Although the merchants manage issuance and redemption, all BTC backing WBTC is held by a single custodian.

### [tBTC v2](https://whitepaper.io/document/691/tbtc-whitepaper)

> tBTC is an open membership system, where the BTC is managed by a rotating set of randomly selected nodes which manage a threshold wallet. The system requires that 51-of-100 randomly selected wallet signers must collaborate to produce a proper signature.

Reviewer 3: Can you explain more about how sBTC differs from tBTC?  In what way(s) is sBTC "better," and how do we know so?

Author: The main difference is in the finality assumptions. tBTC relies on Ethereum's network security, which doesn't recognize Bitcoin. On the other hand, sBTC forks with Bitcoin and inherits finality from Bitcoin. This allows sBTC to provide faster deposit/withdrawal times compared to tBTC (this is described in the SIP).

| yes      | `11111111111mdWK2VXcrA1e7dnvidC` | `SP00000000001WPAWSDEDMQ0B9J72P0KAK2` | `yes-sip-28` | `000000000000000000007965732d7369702d3238` | `OP_DUP` `OP_HASH160` `000000000000000000007965732d7369702d3238` `OP_EQUALVERIFY` `OP_CHECKSIG` |
| no       | `111111111111ACW5wa4RwyeKYEAzMD` | `SP000000000006WVSDEDMQ0B9J73E2TN78`  | `no-sip-28`  | `00000000000000000000006e6f2d7369702d3238` | `OP_DUP` `OP_HASH160` `00000000000000000000006e6f2d7369702d3238` `OP_EQUALVERIFY` `OP_CHECKSIG` |

The addresses have been generated as follows:

- Encode `<message>` in ASCII, with 0-padding.
- Use the resulting `<encoding>` in the Bitcoin script`OP_DUP` `OP_HASH160` `<encoding>` `OP_EQUALVERIFY` `OP_CHECKSIG`.
- The Bitcoin address is the `base58check` of the hash of the Bitcoin script above.
- The Stacks address is the `c32check-encoded` Bitcoin address.

Stackers (pool and solo) vote by sending Stacks dust to the corresponding Stacks address from the account where their Stacks are locked.

Solo stackers only can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.

#### For Non-Stackers:

> Users with liquid STX can vote on proposals using the Ecosystem DAO. Liquid STX is the user’s balance, less any STX they have locked in the PoX stacking protocol, at the block height at which the voting started (preventing the same STX from being transferred between accounts and used to effectively double vote). This is referred to generally as "snapshot" voting.

Reviewer 3: You should link directly to the voting page.

Author: @radicleart do we have the voting page for this?

Author: Added link to the voting page 558713ef63e0c5244dcc32c4248990715d5b148d

### [RBTC](https://rootstock.io/static/a79b27d4889409602174df4710102056/RS-whitepaper.pdf)

**rBTC** is a wrapped BTC asset natively minted on Rootstock, an EVM-compatible sidechain. BTC is secured by a 5-of-9 multi-sig Bitcoin wallet controlled by the Powpeg Federation. Peg operations settle to Bitcoin via merge mining. Instead of collateralizing the system with a new token, peg operators are incentivized by earning a portion of transaction fees. PowPeg operators keep specialized hardware called PowHSMs active and connected to special types of Rootstock full nodes. Since the Bitcoin blockchain and the Rootstock sidechain are not entangled in a single blockchain or in a parent-child relation, peg-in and peg-out transactions require a high number of block confirmations. Peg-ins require 100 Bitcoin blocks, and peg-outs require 4000 Rootstock blocks (roughly 200 Bitcoin Blocks).

This new system shares similarities with existing models but introduces some key distinctions:

> - **Decentralized Custody:** sBTC is secured by a decentralized network of signers rather than a central custodian.

Reviewer 3: Of all three points, this one is the weakest.  `tBTC` and `rBTC` both have a distributed set of signers as well, and per this SIP, sBTC is controlled by a closed set of signers.  Is there something you could add here that further distinguishes sBTC from these two systems?

I wonder if a table would be more succinct.  `?` fields are information I don't know or can't easily find:

| **Feature** | **WBTC** | **tBTC** | **rBTC** | **sBTC** (this SIP) |
| --------------- | -------------- | ------------| ------------ | ------------------------- |
| Spending threshold | 2 of 3 | 51 of 100 | 5 of 9 | 11 of 15 |
| Bitcoin finality | no  |  no  |   yes |  yes |
| Expected Peg-in speed | ? | 3-4 hours | 16 hours | 0.16 hours |
| Expected Peg-out speed | ? | 1 hour | 33.3 hours | 0.16 hours |
| Custodian rotation | no | yes | no | yes |
| Fee structure | % of BTC moved | % of BTC moved | ? | Transaction fees |

Author: Thanks. Updated in 5e3d0de8dcb1530ac7e047115e43485954c401d3

**Authors:**  
- Adriano Di Luzio ([adriano@bitcoinl2labs.com](mailto@adriano@bitcoinl2labs.com))
- Andre Serrano ([andre@bitcoinl2labs.com](mailto:andre@bitcoinl2labs.com))  
- Ashton Stephens ([ashton@trustmachines.co](mailto:ashton@trustmachines.co)) 
- Daniel Jordon ([daniel@trustmachines.co](mailto:daniel@trustmachines.co))
- Friedger Müffke ([friedger@ryder.id](mailto:friedger@ryder.id))  
- Jesus Najera ([jesus@stratalabs.xyz](mailto:jesus@stratalabs.xyz))  
- Joey Yandle ([joey@trustmachines.co](mailto:joey@trustmachines.co))  
- Jude Nelson ([jude@stacks.org](mailto:jude@stacks.org))  
- Mårten Blankfors ([marten@trustmachines.co](mailto:marten@trustmachines.co))  
- Tycho Onnasch ([tycho@zestprotocol.com](mailto:tycho@zestprotocol.com))  

**Consideration:** Governance 

**Type:** Operation  

**Status:** Draft  

**Created:** 2024-06-21

**License:** BSD 2-Clause

**Sign-off:**
- Jason Schrader jason@joinfreehold.com (Governance CAB)
- Jude Nelson jude@stacks.org (Steering Committee)

**Discussions-To:**
- [sBTC Working Group Discussions](https://github.com/stacks-network/sbtc/discussions) 

## Abstract

> This SIP proposes a new wrapped Bitcoin asset, called sBTC, which would be implemented on Stacks 3.0 and later as a SIP-010 token. Stacks today offers a smart contract runtime for Stacks-hosted assets, and the forthcoming Stacks [3.0 release](https://github.com/stacksgov/sips/blob/main/sips/sip-021/sip-021-nakamoto.md) provides lower transaction latency than Bitcoin for Stacks transactions. By providing a robust BTC-wrapping mechanism based on [threshold signatures](https://eprint.iacr.org/2020/852.pdf), users would be able to lock their real BTC on the Bitcoin chain, instantiate an equal amount of sBTC tokens on Stacks, use these sBTC tokens on Stacks, and eventually redeem them for real BTC at 1:1 parity, minus the cost of the relevant blockchain transaction fees.

Author: Thanks for the suggestions. Addressed here. 264382c68009261e1e17be7fe75483bf58deacba

This is the first of several SIPs that describe such a system. This SIP describes the threshold signature mechanism and solicits from the ecosystem both a list of signers and the criteria for vetting them. These sBTC signers would be responsible for collectively holding all locked BTC and redeeming sBTC for BTC upon request. Given the high-stakes nature of their work, the authors of this SIP believe that such a wrapped asset can only be made to work in practice if the Stacks ecosystem members can reach broad consensus on how these signers are chosen. Thus, the first sBTC SIP put forth for activation concerns the selection of sBTC signers.

This SIP outlines but does not describe in technical detail the workings of the first sBTC system. A separate SIP will be written to do so if this SIP successfully activates.

> 

Author: Thanks for the suggestions. Addressed here. 264382c68009261e1e17be7fe75483bf58deacba

## Introduction

### Glossary

| Term                | Definition                                                                                                                                                            |
|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **SIP-10 Token**    | A token on the Stacks blockchain that adheres to the fungible token standards outlined in [SIP-10](https://github.com/stacksgov/sips/blob/main/sips/sip-010/sip-010-fungible-token-standard.md).                                                                       |

> | **sBTC**            | A SIP-10 token on the Stacks Blockchain that can be turned back into BTC on the Bitcoin Blockchain. 1 sBTC is equivalent to 1 BTC on the Bitcoin Blockchain.           |

Reviewer 5: "Bitcoin blockchain" (lower b) used elsewhere in the doc, pick one or the other for consistency

| **sBTC Peg Wallet** | The single UTXO holding the entire BTC balance that’s pegged into sBTC. This peg wallet is managed and maintained by the sBTC Signers.                                  |
| **Stacks Signer**   | An entity that receives PoX payouts for stacking their STX tokens and actively participating in the Stacks protocol by signing mined blocks.                            |
| **sBTC Signer**     | An entity that will sign sBTC operations and communicate with contracts on the chain to make that feasible. This entity has partial access to spending the sBTC UTXO.   |
| **sBTC Signer Set** | The set of all sBTC signers. Each is registered with the .sbtc contract and these entities as a group collectively maintain the sBTC's Bitcoin UTXO.      |
| **sBTC Signer API** | An API exposed by the sBTC Signer that handles basic low-level commands.                                                                                                |
| **Deposit API**     | A third-party API that communicates with the sBTC Signers via the sBTC Signer API.                                                                                      |

> 

Author: Thanks, added to the glossary 69d40a5f4f0ad98eb448ba44e7c31ca054820aa3

The sBTC protocol not only addresses the limitations of Bitcoin's scripting system but also provides a secure and decentralized solution for utilizing Bitcoin in various applications.

## Design

While the first sBTC implementation is under development, the wrapped nature of the sBTC token means that any such system would have the following properties:

- sBTC is a SIP-10 token backed 1:1 by BTC.
- The sBTC peg wallet is maintained by the set of sBTC signers. These signers are responsible for the security and maintenance of the wallet, ensuring that sBTC is redeemable for BTC.
- Bitcoin can be converted into sBTC within 3 Bitcoin blocks, and sBTC can be converted into Bitcoin within 6 Bitcoin blocks. sBTC relies on the forking behavior guaranteed by [SIP-021](https://github.com/stacksgov/sips/blob/main/sips/sip-021/sip-021-nakamoto.md) in order to maintain the peg wallet correctly across forks.

## Specification

> Management of the sBTC peg wallet on the Bitcoin blockchain shall be managed by the proposed set of signers through a democratic process, involving the sBTC Signer Set rather than a single custodian. At launch, the sBTC protocol will be maintained by 15 independent entities that make up the sBTC Signer Set and each unique signer is allocated exactly one vote. The system requires at least 70%, or 11 out of 15 signatures, for an sBTC operation to be fulfilled. The eligibility criteria to become an sBTC Signer are determined through the community governance process of ratifying this SIP. For this release, sBTC will not be part of Stacks consensus.

Reviewer 5: So just to confirm, if 2 signers STX power are huge, holding than 70% number of STX in the signer set, it would still require 11 individual votes for the transaction to approve?

Reviewer 2: i don't think the sbtc signers will be voting based on any "STX power", since that would be a different signer binary they run (or not. i think the eligibility is that they do run one, but it's not relevant for sbtc signer set voting)

Author: Correct, it is one signer one vote.

> 

Reviewer 6: This section could benefit from a couple diagrams. I will suggest three

Diagram 1: 11-of-15 (>=70%) Threshold Signature Setup
````
+------------+  
|  Signer 1  |  
+------------+  
+------------+  
|  Signer 2  |  
+------------+  
+------------+  
|  Signer 3  |  
+------------+  
      .
      .
      .
+------------+                  +--------------------+
| Signer 15  |----(11-of-15)---->|  Shared Multi-Sig  |
+------------+                  |    BTC Wallet      |
                                 +--------------------+
````

sBTC Signers are responsible for accepting or rejecting all sBTC deposit and withdrawal operations submitted to the network. For a transaction to be fulfilled, at least 70% of the signers need to approve the transaction. This means that the liveness and reliability of the signers is crucial to the success of the protocol. The system is live ("resilient") if at least 70% of the sBTC Signer voting power are online and honest. Then (and only then), deposits and withdrawals happen in a timely manner. The system is safe ("trustworthy") if at least 30% of the sBTC Signer voting power is honest. Then, no theft of funds can occur. Additionally, more details on sBTC deposit and withdrawals are included in the appendix of this SIP. 

> 

Reviewer 6: Diagram 2: sBTC Minting process (peg-in)
````
+-----------+               +-------------+             +-----------+
|   User    |--Lock BTC----->|  Signers    |--Confirm--> | Mint sBTC |
+-----------+               | (Multi-Sig) |             +-----------+
                             +-------------+                 
````

Diagram 3: sBTC Redemption Process (peg-out)
````
+-----------+               +-------------+             +-----------+
|   User    |--Burn sBTC---->|  Signers    |--Unlock BTC->| Receive   |
|           |               | (Multi-Sig) |             |   BTC      |
+-----------+               +-------------+             +-----------+
````

Author: We had these diagrams in an earlier version of the SIP, but decided it was out of scope for the signer criteria, so we removed them. Thanks for the suggestion though!

### sBTC Signer Responsibilities
The sBTC signers play a critical role in the security and operations of the sBTC system. Their responsibilities can be grouped into two categories: tasks mandated by the sBTC protocol and operational best practices to effectively manage the sBTC system.

**Protocol-Mandated Tasks:**
- Signers must accept and process BTC deposit requests.
- They must fulfill BTC withdrawal requests in a timely manner, ensuring accurate execution.
- Signers are responsible for moving BTC to a new UTXO when private keys are rotated.

> - Signers must perform UTXO consolidation as BTC is deposited to optimize the number of unspent outputs [1].

Reviewer 7: Yes, the consolidation fee is specified explicitly by the depositor actually; I've just suggested a change in wording:

https://github.com/stacksgov/sips/pull/186/files#r1815536806

The deposit request on Bitcoin contains data about how much of their deposit a depositor is willing to have go to their fee. Every deposit pays the same fee to consolidate so having a very low `max-fee` won't help you if the fee is too low. 

Part of the benefit of this is that there'll be no hidden fees later to support sweeping up a bunch of UTXOs, all the fees necessary to facilitate the deposit are done right around deposit time.

> - Signers must perform UTXO consolidation as BTC is deposited to optimize the number of unspent outputs [1].

Reviewer 2: merged suggestion

The following eligibility criteria will be used to identify the sBTC Signers:

- Does the proposed sBTC signer have a demonstrable operating history which shows their experience and reliability in running blockchain services?
- Has the proposed sBTC signer participated in running a Stacks signer instance on Stacks 2.5 testnet or mainnet, and can they provide metrics showing this (ex: amount of stacks stacked over past several cycles)?
  * Note: The sBTC signer is a Stacks event observer, meaning that the experience of running a Stacks node signer directly translates to running an sBTC signer.
- Does the proposed sBTC signer agree to use reasonable efforts to maintain >99% uptime on the sBTC Signer? 
  * Note: This metric may be self-affirmed if independent verification is not possible, or confirmed by on-chain voting/stacking activity.
- Does the proposed sBTC signer commit to a direct communication channel to be set up with the sBTC core engineers in order to respond to urgent updates within 24 hours?

> - Has the signer made contributions to Bitcoin or the Stacks network over the past year that demonstrate their commitment to the growth and success of the network? Examples include, but are not limited to: publishing independent research, marketing, co-authoring a SIP, submitting a Stacks pull request/issue, providing feedback on Stacks core development, or contributing to a Stacks Working Group.

Reviewer 5: I'm sure this has already been considered, but it might be worth explicitly stating 'geographical distribution' as a criterion for the overall set of signers criteria/consideration. Without this, it's possible that each of the 15 signers could meet individual eligibility criteria, but the entire group could still be U.S.-based, which could pose a jurisdictional risk.

Author: Yes, thanks Hiro. This should have been included (it was one of the factors we've considered). I'll update the SIP.

Author: resolved 02df3fde8c1d84c0a280a120edda322285503ee4

> The criteria described above will be used to identify sBTC Signers that are able to meet some or all of the responsibilities described in the previous section.

Reviewer 5: Should the list of signers be shown in the SIP-028? 
Is feedback on the signer set for rest of stacks community? https://github.com/stacks-network/sbtc/discussions/624 
As people may miss the opportunity to feedback if it's not on social or this SIP text
OR be in a file in the same folder as the SIP?

### Selection Process
The sBTC Signer Set will be finalized from the list of eligible Signers, based on the above criteria. The [sBTC Working Group](https://github.com/orgs/stacks-network/discussions/508) will conduct the vetting process and the results will be published as a discussion in the [sBTC Github repository](https://github.com/stacks-network/sbtc/discussions/624). 

The selection process is as follows:
1. **Nomination Phase**: Open a call for nominations within the community.
2. **Evaluation & Community Feedback**: The proposed signer set will be published to provide transparency.
3. **SIP Vote**: The community will vote on the sBTC signer criteria.
4. **Final Selection**:  If SIP-028 is ratified, then the proposed signer set voted upon in step 3 shall be the initial signing set for sBTC. If the signer set changes during the vote, such as by the withdrawal of one or more candidates, then the vote will be restarted in a subsequent reward cycle (to be determined if this comes to pass).

   
### Updating The sBTC Signer Set

> In the event that the sBTC Signer Set needs to be updated (for example, if a signer is no longer available to complete their responsibilities) sBTC Signers can perform a threshold vote to agree on the updated set. This process will also be performed if a signer needs to rotate their cryptographic keys.

Reviewer 5: is the threshold same as the threshold for approving pegging transactions? if yes best to state?
As it does not necessarily have to be the same.

The following table summarizes the main design differences between these systems:

| Feature                | WBTC            | tBTC             | rBTC             | sBTC (this SIP)    |
|------------------------|-----------------|------------------|------------------|--------------------|
| Spending threshold      | 2 of 3          | 51 of 100        | 5 of 9           | 11 of 15           |
| Bitcoin finality        | No              | No               | Yes              | Yes                |
| Expected Peg-in speed   | 1 hour          | 1-3 hours        | 16 hours         | 0.5 hours          |
| Expected Peg-out speed  | 1 hour          | 3-5 hours        | 33.3 hours       | 1 hour             |
| Custodian rotation      | No              | Yes              | No               | Yes                |
| Fee structure           | % of BTC moved  | % of BTC moved   | Transaction fees | Transaction fees   |

In conclusion, the sBTC system shares similarities with existing models but introduces some key distinctions:

> - **Bitcoin Finality:** sBTC inhereits Bitcoin finality from [Stacks 3.0](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md), which ensures that sBTC transactions receive the same level of security provided by the Bitcoin network.

Reviewer 5: inherits

Reviewer 2: resolved

> - **Bitcoin Finality:** sBTC inhereits Bitcoin finality from [Stacks 3.0](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md), which ensures that sBTC transactions receive the same level of security provided by the Bitcoin network.

Reviewer 2: ```suggestion
- **Bitcoin Finality:** sBTC inherits Bitcoin finality from [Stacks 3.0](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md), which ensures that sBTC transactions receive the same level of security provided by the Bitcoin network.
```

- **Faster Deposit & Withdrawal Times:** sBTC enables BTC withdrawals without the long delays associated with block confirmations in other systems. This is achieved through the finality rules described in [Stacks 3.0](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md).

## Activation

> sBTC is designed to activate on Stacks 3.0 as defined in [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md). Therefore, this SIP is only meaningful when SIP-021 activates. The sBTC Working Group plans to observe at least 2-4 weeks of network behavior on Stacks Nakamoto to ensure a stable release. After this period, sBTC can be activated on the Stacks network without requiring a separate hard fork.

Reviewer 5: activate "on" or "after"?

### Process of Activation

Users can vote to approve this SIP with either their locked/stacked STX or with unlocked/liquid STX, or both. The SIP voting page can be found at [sbtc.vote](https://sbtc.vote). The criteria for the stacker and non-stacker voting is as follows.

#### For Stackers:

In order for this SIP to activate, the following criteria must be met by the set of Stacked STX:

> - At least 80 million Stacked STX must vote, with least 66% (48 million) voting "yes".

Reviewer 5: This SIP seems important enough to get high consensus of 80%. 
66% required for Stacked STX we have not done before.

So perhaps think about using below to match other SIPs.

* At least 80 million Stacked STX must vote at all to activate this SIP.
* Of the Stacked STX that vote, at least 80% of them must vote "yes."

And update non-Stackers vote power %  to 80% to match.

Reviewer 2: @andrerserrano @aldur

Author: Updated to 70% to match the sBTC signer operations.

- Use the resulting `<encoding>` in the Bitcoin script`OP_DUP` `OP_HASH160` `<encoding>` `OP_EQUALVERIFY` `OP_CHECKSIG`.
- The Bitcoin address is the `base58check` of the hash of the Bitcoin script above.
- The Stacks address is the `c32check-encoded` Bitcoin address.

Stackers (pool and solo) vote by sending Stacks dust to the corresponding Stacks address from the account where their Stacks are locked.

Solo stackers only can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.

#### For Non-Stackers:

Users with liquid STX can vote on proposals directly at [sBTC.vote](https://sbtc.vote) using the Ecosystem DAO. Liquid STX is the user’s balance, less any STX they have locked in the PoX stacking protocol, at the block height at which the voting started (preventing the same STX from being transferred between accounts and used to effectively double vote). This is referred to generally as "snapshot" voting.

For this SIP to pass, 66% of all liquid STX committed by voting must be in favor of the proposal. This precedent was set by [SIP-015](https://github.com/stacksgov/sips/blob/main/sips/sip-015/sip-015-network-upgrade.md).

> The act of not voting is the act of siding with the outcome, whatever it may be. We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

Reviewer 5: "The act of not voting is the act of siding with the outcome" - @wileyj had issue with this wording on previous sip, might wanna check with him on this.

Reviewer 2: correct - it's not relevant, and the assumption is incorrect  regardless.
I don't like the wording or the implication, but it's not a blocker to moving to a vote.

Author: Agreed, this has been removed.

# Preamble

**SIP Number:** 028  

**Title:** Signer Criteria for sBTC, A Decentralized and Programmable Asset Backed 1:1 with BTC  

**Authors:**  
- Adriano Di Luzio ([adriano@bitcoinl2labs.com](mailto@adriano@bitcoinl2labs.com))
- Andre Serrano ([andre@bitcoinl2labs.com](mailto:andre@bitcoinl2labs.com))  
- Ashton Stephens ([ashton@trustmachines.co](mailto:ashton@trustmachines.co)) 
- Daniel Jordon ([daniel@trustmachines.co](mailto:daniel@trustmachines.co))
- Friedger Müffke ([friedger@ryder.id](mailto:friedger@ryder.id))  
- Jesus Najera ([jesus@stratalabs.xyz](mailto:jesus@stratalabs.xyz))  
- Joey Yandle ([joey@trustmachines.co](mailto:joey@trustmachines.co))  
- Jude Nelson ([jude@stacks.org](mailto:jude@stacks.org))  
- Mårten Blankfors ([marten@trustmachines.co](mailto:marten@trustmachines.co))  
- Tycho Onnasch ([tycho@zestprotocol.com](mailto:tycho@zestprotocol.com))  

**Consideration:** Governance 

**Type:** Operation  

**Status:** Draft  

**Created:** 2024-06-21

**License:** BSD 2-Clause

**Sign-off:**

> - Andre Serrano andre@bitcoinl2labs.com (SIP Editor)

Reviewer 5: Delete "Andre Serrano andre@bitcoinl2labs.com (SIP Editor)" line

Author: Done 2dfc68d59e67ffc85e359e2f1806658638699ce1

> - Jude Nelson jude@stacks.org (Steering Committee)

Reviewer 3: I haven't signed off on anything

For this SIP to pass, 66% of all liquid STX committed by voting must be in favor of the proposal. This precedent was set by [SIP-015](https://github.com/stacksgov/sips/blob/main/sips/sip-015/sip-015-network-upgrade.md).

The act of not voting is the act of siding with the outcome, whatever it may be. We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

## Appendix
[1] https://github.com/stacks-network/sbtc/issues/52

[2] https://github.com/stacks-network/sbtc/pull/186

[3] https://github.com/stacks-network/sbtc/issues/30

### Specification

#### Deposits

The main steps of the sBTC Deposit flow will be as follows:

1. **Deposit request:** A bitcoin holder creates a transaction on Bitcoin.
  - The deposit transaction contains a UTXO (deposit UTXO) spendable by sBTC Signers, with an OP_DROP payload.

>   - The payload contains the recipient address of the sBTC among other relevant info for the deposit.
>     - The relevant info could contain a fee suggestion or max_fee.

Reviewer 7: ```suggestion
  - The payload contains the recipient address of the sBTC and he maximum fee the depositor is willing to have go towards the consolidation of the deposits into a single UTXO.
```

- The Bitcoin address is the `base58check` of the hash of the Bitcoin script above.
- The Stacks address is the `c32check-encoded` Bitcoin address.

Stackers (pool and solo) vote by sending Stacks dust to the corresponding Stacks address from the account where their Stacks are locked.

Solo stackers only can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.

#### For Non-Stackers:

Users with liquid STX can vote on proposals directly at [sBTC.vote](https://sbtc.vote) using the Ecosystem DAO. Liquid STX is the user’s balance, less any STX they have locked in the PoX stacking protocol, at the block height at which the voting started (preventing the same STX from being transferred between accounts and used to effectively double vote). This is referred to generally as "snapshot" voting.

For this SIP to pass, 70% of all liquid STX committed by voting must be in favor of the proposal. 

We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

> 

Reviewer 5: Submitting the vote results @obycode.
1. While reviewing the raw transaction data from the CSV file, I identified a minor discrepancy, resulting in a 0.01% overall shift in favor of YES votes. 
2. This discrepancy stems from a 247K STX reduction in voting power within the Unlocked STX track compared to the figures originally shared in the community blog post (https://stacks.org/sip-028-approved). I will reach out to the Foundation to request an update to the blog.

If anyone wants to triple-check, the CSV file can be retrieved from the voting site [here](https://stx.eco/dao/proposals/SP3JP0N1ZXGASRJ0F7QAHWFPGTVK9T2XNXDB908Z.sip028-signer-criteria-for-sbtc/results2?chain=mainnet).

@andrerserrano, if you could incorporate below update into your file, that would be great—then we can finally get this PR merged.

# Activation Status
At the end of Bitcoin block height 869,749, the following vote was recorded: **A total of 135,940,943 STX participated, with 135,801,729 STX (99.99%) voting 'Yes' and 139,214 STX (0.10%) voting 'No'.**

A breakdown of the vote is as follows:
- For STX locked in stacking, a total of 126,084,232 STX participated. Of this, 125,945,571 STX (99.89%) from 77 accounts voted 'Yes', while 138,661 STX (0.11%) from 1 account voted 'No'.
- For unlocked STX, a total of 9,718,049 STX participated. Of this, 9,717,496 STX (99.99%) from 1,175 accounts voted 'Yes', while 553 STX (0.01%) from 40 accounts voted 'No'.

All voting criteria from STX holders have been met. A breakdown of the transactions can be found [here](https://stx.eco/dao/proposals/SP3JP0N1ZXGASRJ0F7QAHWFPGTVK9T2XNXDB908Z.sip028-signer-criteria-for-sbtc/results2).

The API where results come from can be found [here](https://github.com/radicleart/stxeco-api-vote).
